.github/workflows/tests.yml

```diff
@@ -12,18 +12,16 @@ jobs:
 
     strategy:
       fail-fast: false
-      matrix:
-        go: ['1.20', '1.19', '1.18', '1.17']
```

Comment:
I think we should leave the matrix in here. We'll want to run on Go 1.21 when it comes out (in the next week or two?).

Reply:
Left comments on Mattermost, but I'll copy them here for the record.

I don't see why it's necessary to run unit tests with multiple Go versions. If I'm interpreting [this](https://go.dev/ref/mod#go-mod-file-go) right, running the tests with 1.21 will just use the 1.20 semantics, since that is the version declared in go.mod. In which case there is no reason to run the Go tests with 1.21 as well.

Reply:
Just to be clear, I meant leave the matrix in but with just `['1.20']` in it, so we can easily add 1.21 later.

I think it's still useful to run on both supported versions. As I understand it, Go 1.21 will attempt to run with 1.20 semantics for language features and APIs, but when 1.21 comes out, running the tests on Go 1.21 as well as 1.20 will ensure that it definitely does run and pass all the tests on both. Even if it's just a matter of error messages being different in the two versions or something where our tests are too specific, it would be good to catch.

That's my take, anyway, though I don't feel particularly strongly about it.

Reply:
> Even if it's just a matter of error messages being different in the two versions or something where our tests are too specific, it would be good to catch.

I just can't really see that happening, due to the Go compatibility promise. The one exception would be the [new loop semantics](https://github.com/golang/go/discussions/56010), but that's easy enough to test manually.

IMO we should test using the same version(s) of Go we build Pebble with. If we only build Pebble binaries using a single Go version, then we only need to test with that version.

So I don't see the benefit to running tests with multiple Go versions, but I'm interested to see what others think. @hpidcock @jnsgruk @flotter @anpep @tlm (@ anyone else) ?

Reply:
> IMO we should test using the same version(s) of Go we build Pebble with. If we only build Pebble binaries using a single Go version, then we only need to test with that version.

That's fair enough -- I'd be fine with that. [Looks like](https://github.com/canonical/pebble/blob/6991b3c387ed673436afe6e7afea73f64b69d4ed/.github/workflows/binaries.yml#L30) we build binaries with the latest stable version of Go (though arguably we should explicitly pin a version there). Interested to hear what others think too.

Reply:
I'd prefer to stick with the matrix, but trim to just `['1.20']`.

We will likely not bump the actual go version of pebble to 1.21 immediately, but we will probably want to test on 1.21.

@barrettj12 if you can just change this to a single version for now, that would be handy.

While you're in there, can you adjust the binary builder to use the version of go from `go.mod`?

Reply:
I believe you can do that like so:

```yaml
- uses: actions/setup-go@v4
    with:
      go-version-file: "go.mod"
```